### ISSUE TITLE
Clarity buffer types not decoded properly

### ISSUE THREAD

**Issue by aulneau** [stx-labs/stacks.js#834]
See https://github.com/blockstack/explorer/issues/246, it seems that clarity types are not being decoded correctly within the sidecar, which is using the transactions-lib to do so. cc @yknl @reedrosenbluth

**Comment by yknl** [comment]
Can you provide an example of the data from the API or link to a transaction that is decoded incorrectly?

**Comment by aulneau** [comment]
@yknl https://testnet-explorer.blockstack.org/txid/0x606c308b63701ce6731b2f6104d3186f7fbc0e0307a80b4f2c02acdd849fb69b

see function summary

**Comment by zone117x** [comment]
Buffer values are being decoded as unicode rather than into hex strings, see:
```
(tuple (hashbytes "O\u000e\u0010E/\u00135E'8^P\u0002Wf7gyy") (version 0x00))
```
And corresponding ABI:
```
(pox-addr (tuple (version (buff 1)) (hashbytes (buff 20))))
```

**Comment by reedrosenbluth** [comment]
@zone117x  if you are using the `cvToString()` method, there is an optional second argument to specify the buffer `toString()` encoding. It tries to decode the buffer as ascii by default, but if you input `hex` as the second arg, it will decode as hex.

`function cvToString(val: ClarityValue, encoding: 'tryAscii' | 'hex' = 'tryAscii'): string`

**Comment by zone117x** [comment]
I think now that we have have first class string support, we should probably always default to hex, and possibly remove the try/catch encoding logic we had initially as a workaround.

**Comment by yknl** [comment]
Sounds like switching the default option on `cvToString()` to hex is the solution here.

**Comment by reedrosenbluth** [comment]
https://github.com/blockstack/blockstack.js/pull/838

**Comment by friedger** [comment]
fixed by #838 ?
